doc/cprover-manual/memory-analyzer.md

```diff
@@ -0,0 +1,78 @@
+[CPROVER Manual TOC](../../)
+
+## Memory Analyzer
+
+The memory analyzer is a front-end for running and querying GDB in order to
+obtain a state of the input program. A common application would be to obtain a
+snapshot of a program under analysis at a particular state of execution. Such a
+snapshot could be useful on its own: to query about values of particular
+variables. Furthermore, since that snapshot is a state of a valid concrete
+execution, it can also be used for subsequent analyses.
+
+## Usage
+
+We assume that the user wants to inspect a binary executable compiled with
+debugging symbols and a symbol table information understandable by CBMC, e.g.
+(having `goto-gcc` on the `PATH`):
+
+```sh
+$ goto-gcc -g -std=c11 input_program.c -o input_program_exe
+```
+
+Calling `goto-gcc` instead of simply compiling with `gcc` produces an ELF binary
+with a goto section that contains the goto model (goto program + symbol table).
```

Comment:
Hmm, do we not document this anywhere else? If we don't, that's a problem, else we should have a cross reference here.

Reply:
There is a (somewhat terse) description in `doc/cprover-manual/goto-cc-variants.md`.

Reply:
Even if terse, it seems like the right place for such documentation. So please do add a reference to that bit of documentation, in addition to what you are documenting about it here.